doc/cephfs/disaster-recovery-experts.rst

```diff
@@ -204,63 +204,99 @@ Using an alternate metadata pool for recovery
-results to the alternate pool:
+Now perform the recovery of the metadata pool from the data pool:
 
 ::
 
-    cephfs-data-scan scan_extents --alternate-pool recovery --filesystem <original file system name> <original data pool name>
-    cephfs-data-scan scan_inodes --alternate-pool recovery --filesystem <original file system name> --force-corrupt --force-init <original data pool name>
-    cephfs-data-scan scan_links --filesystem recovery-fs
+    cephfs-data-scan init --force-init --filesystem cephfs_recovery --alternate-pool cephfs_recovery_meta
+    cephfs-data-scan scan_extents --alternate-pool cephfs_recovery_meta --filesystem <fs_name> <data_pool>
+    cephfs-data-scan scan_inodes --alternate-pool cephfs_recovery_meta --filesystem <fs_name> --force-corrupt <data_pool>
+    cephfs-data-scan scan_links --filesystem cephfs_recovery
+
+.. note::
+
+   The scan procedures above go through the entire data pool. This may take a
+   significant amount of time. See the previous section how to distribute this
+   task among workers.
 
 If the damaged file system contains dirty journal data, it may be recovered next
 with:
 
 ::
 
-    cephfs-journal-tool --rank=<original filesystem name>:0 event recover_dentries list --alternate-pool recovery
-    cephfs-journal-tool --rank recovery-fs:0 journal reset --force
+    cephfs-journal-tool --rank=<fs_name>:0 event recover_dentries list --alternate-pool cephfs_recovery_meta
+    cephfs-journal-tool --rank cephfs_recovery:0 journal reset --force
 
 After recovery, some recovered directories will have incorrect statistics.
-Ensure the parameters mds_verify_scatter and mds_debug_scatterstat are set
-to false (the default) to prevent the MDS from checking the statistics, then
-run a forward :doc:`scrub </cephfs/scrub>` to repair them. Ensure you have an
-MDS running and issue:
+Ensure the parameters ``mds_verify_scatter`` and ``mds_debug_scatterstat`` are
+set to false (the default) to prevent the MDS from checking the statistics. Next,
+start the MDS:
 
 ::
 
-    ceph tell mds.a scrub start / recursive repair
+    ceph fs set cephfs_recovery joinable true
 
-.. note::
+Then run a forward :doc:`scrub </cephfs/scrub>` to repair the statistics.
+Ensure you have an MDS running and issue:
 
-    In Nautilus and above versions, tell interface scrub command is preferred
-    than scrub_path. For older versions only scrub_path asok command is
-    supported. Example::
+::
+
+    ceph fs status # get active MDS
+    ceph tell mds.<id> scrub start / recursive repair
+
+It is recommended to migrate any data from the recovery file system as soon as
+possible. Do not restore the old file system while the recovery file system is
+operational.
+
+.. note::
 
-        ceph daemon mds.a scrub_path / recursive repair
+    If the data pool is also corrupt, some files may not be restored because
+    backtrace information is lost. The data for files will also likely have
+    holes corresponding to the PGs that were lost.
```

Comment:
s/The data.*/If any data objects are missing (due to issues like lost data pool PGs), the recovered files will contain holes in place of the missing data./